exo/inference/mlx/sharded_utils.py

```diff
@@ -128,18 +123,7 @@ def load_model_shard(
             class_predicate=None,
         )
 
-    filtered_weights = {}
-    for k, v in weights.items():
-        if k.startswith("model.layers."):
-            layer_num = int(k.split('.')[2])
-            if shard.start_layer <= layer_num <= shard.end_layer:
-                new_key = f"model.layers.{layer_num - shard.start_layer}." + '.'.join(k.split('.')[3:])
-                filtered_weights[new_key] = v
-        else:
-            filtered_weights[k] = v
-    weights = filtered_weights
-
-    model.load_weights(list(weights.items()), strict=False)
+    model.load_weights(list(weights.items()))
```

Comment:
Again, the part you removed is important to ensure that only the weights for this shard are loaded into memory.

Reply:
It's okay as it's lazy loaded.